v1/charm-dev.yaml

```diff
@@ -88,33 +98,48 @@ instances:
           lxd init --auto
           adduser ubuntu lxd
 
+        - |
+          # Make sure juju directory is there
+          # https://bugs.launchpad.net/juju/+bug/1995697
+          sudo -u ubuntu mkdir -p /home/ubuntu/.local/share/juju
+
         - |
           # setup microk8s and bootstrap
-          adduser ubuntu microk8s
+          usermod -a -G snap_microk8s ubuntu
+          usermod -a -G microk8s ubuntu
           microk8s status --wait-ready
-          
+
           microk8s.enable metrics-server
           microk8s.kubectl rollout status deployments/metrics-server -n kube-system -w --timeout=600s
 
           # The dns addon will restart the api server so you may see a blip in the availability
-          # Separating addons to avoid errors such as 
+          # Separating addons to avoid errors such as
           # dial tcp 127.0.0.1:16443: connect: connection refused
-          microk8s.enable dns
+          microk8s.enable dns rbac
           microk8s.kubectl rollout status deployments/coredns -n kube-system -w --timeout=600s
-          
+
+          # wait for storage become available
           microk8s.enable hostpath-storage
-          microk8s.enable ingress
-          # wait for storage, ingress to become available
           microk8s.kubectl rollout status deployments/hostpath-provisioner -n kube-system -w --timeout=600s
-          microk8s.kubectl rollout status daemonsets/nginx-ingress-microk8s-controller -n ingress -w --timeout=600s
 
-          sudo -u ubuntu juju bootstrap --no-gui microk8s charm-dev
-          sudo -u ubuntu juju add-model --config logging-config="<root>=WARNING; unit=DEBUG" --config update-status-hook-interval="60m" welcome
+          # MetalLB
+          IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+          microk8s enable metallb:$IPADDR-$IPADDR
+          microk8s.kubectl rollout status daemonset.apps/speaker -n metallb-system -w --timeout=600s
+
+          # bootstrap controllers
+          sudo -u ubuntu juju bootstrap localhost lxd
+          sudo -u ubuntu juju add-model --config logging-config="<root>=WARNING; unit=DEBUG" --config update-status-hook-interval="60m" welcome-lxd
+          sudo -u ubuntu juju bootstrap microk8s microk8s
+          sudo -u ubuntu juju add-model --config logging-config="<root>=WARNING; unit=DEBUG" --config update-status-hook-interval="60m" welcome-k8s
```

Comment:
JFMI, why do we change default update-status-hook-interval (5m->60m)?

Reply:
This is a dev blueprint and `update-status` often interferes with charm logic.
For example, a bundle deployment should reach active/idle without relying on update-status interval.
For startup, the 60m is a stand-in for "Inf". For long running VMs, it is still nice to have that hook fire some times.
Obviously this is something you can change after launch, but I found it to be quite useful as a default.

Reply:
I have a different opinion. Unfortunately `update-status` used as a periodic cron job/event to make some checks/self-fixes. Increasing it so high means they will never be executed.

I agree with you on popular race-conditions with `update-status` during the application bootstrap.
Still I prefer to see Juju defaults and fix the problematic charms (as users will those issues outside of charmed-dev anyway). Muting them here produces false-positive feelings. My $0.05 here.

Anyway, it is merged. Thank you for 3.1 fix!

Reply:
> `update-status` used as a periodic cron job/event to make some checks/self-fixes

Yep, this is an unfortunate reality, but hopefully the upcoming pebble notices will replace (the unreliable) reliance on update-status interval.

> Muting them here produces false-positive feelings.

I meant the opposite actually: having a short update-status interval _helps_ resolve charm bugs, which is what imo gives false positive feelings.